0004-market-integration-token-standard.md

```diff
@@ -65,38 +80,43 @@ At time of writing, this standard is established with several constraints found
     unimplemented();
   }
 
-  export function getToken(tokenId:string):Token {
+  // returns TokenTypeID, which is also an index
+  export function mintTokenType(name:string, totalSupply:u64):u32 {
     unimplemented();
   }
 
-  export function getTokenOwner(tokenId:string):string {
+  export function getToken(tokenId:u64):Token {
     unimplemented();
   }
 
-  export function getTokenTypes():TokenType[] {
+  export function getTokenType(tokenTypeId:u32) {
     unimplemented();
   }
-  
-  export function getTokensByOwner(ownerId:string, startTokenId:string = null, limit:u32 = 10):Token[] {
+
+  export function getTokenOwner(tokenId:u64):string {
     unimplemented();
   }
 
-  export function getCountByOwner(tokenType: TokenType, ownerId: string) {
+  export function getAllTokenTypes():TokenType[] {
     unimplemented();
   }
-  
-  export function getTokenData(tokenId:string):bytes {
+
+  export function getTokensByOwner(ownerId:string, startTokenId:u64 = 0, limit:u32 = 10):Token[] {
+    unimplemented();
+  }
+
+  export function getCountByOwner(ownerId: string):u64 {
     unimplemented();
   }
 
-  export function transfer(recipientId:string, tokenIds:string[], onTransfer:PayloadCallback = null):void {
+  export function transfer(recipientId:string, tokenIds:u64[], onTransfer:PayloadCallback = null):void {
     unimplemented();
```

Comment:
Will this not work better as an overloaded function (allowing for optimised transfers when `tokenIds.length === 1`)?
In terms of `onTransfer`, would a default of `() => {}` or `no-op` be clearer (since `null` would imply that you would need to do null checking, instead of unconditionally applying the cb.

Reply:
This also is one `user`, many tokens.
Is there scope for many `user` + many tokens (much like a `zip` + `map`)

Is the onTransfer called once? After the first transfer? After the last?

Reply:
The idea of overloading is a good idea, but is not yet supported in AssemblyScript. This is likely going to come soon.

Reply:
Also, onTransfer will be called each transfer. Transfers happen as batches. This is not truly a callback, since it may be called before the actual transfer occurs.